src/library/arith_instance_manager.h

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Daniel Selsam
+*/
+#pragma once
+#include "kernel/expr.h"
+#include "library/type_context.h"
+
+namespace lean {
+
+class arith_instance_info {
+    type_context * m_tctx_ptr;
+
```

Comment:
This looks wrong. `type_context` is usually a transient object that lives on the stack.
`arith_instance_info` is stored in thread local storage, right?